# Hi, I'm Shasank Shekhar Padhi 

AI/ML Engineer | Knowledge Graph Architect | Bioinformatics Scientist   
Senior Research Associate at Syngene | Bengaluru, India 

## About me
A Data Science/Machine Learning professional with over 3 years of experience in AI/ML model development, data pipelines, and knowledge graph applications. Skilled in GenAI, NLP, LLM and building knowledge graph solutions to drive data-driven insights, enhance decision-making, and deliver scalable solutions through cross-functional collaboration across diverse industries.

## Technical skills
`Programming languages`: Python | R | Cypher | Shell Scripting  
`AI/ML`- Frameworks: PyTorch | Scikit-learn | TensorFlow | SciPy | LLM | GNN | GEN-AI | RAG   
`Deployment`: Docker | GIT | RESTAPI | LangChain  
`Databases`: Neo4J | MongoDB | ChromaDB  
`Bioinformatics`: RNA seq, scRNA seq, WGS | GWAS | Systems Modeling  

## Highlights
1. Antibody generation:
Built an end-to-end Generative AI pipeline for antibody sequence generation and optimization using LLMs ensuring high-affinity binders.
Impact: Reduced discovery time compared to traditional methods.

2. Biomedical Knowledge Graph (Neo4j):
Constructed a ~10M relationship KG from biological, clinical, and real-world data (RWD), harmonized with controlled vocabularies for each entity.
Application included drug repurposing, target identification, safety assessment for toxicity and organ wise stratification.
Impact: Reduced months of work to weeks by employing link prediction, node classification and community detection algorithms.

3. Automated ML model development for Drug Properties:
Developed an AutoML pipeline for QSPR modeling developing predictive models. Enabled non-ML researchers to train and evaluate models without prior coding experiance.
Impact: Reduced dependency on data scientists.

4. Biomarker Discovery (Ensemble ML + Omics)
Trained XGBoost models on RNA-seq/scRNA-seq data for biomarker identification.
Identified novel biomarkers for patient stratification.

5. VectorDB for Drug Binding Sites:
Developed a pipeline to accelerate searches across a vector database containing binding pockets, enabling rapid identification of similar sites to assess target protein and possible adverse events

6. Temporal KG for Diabetes EHR (Neo4j + LLaMA3 RAG):
Built a time-aware knowledge graph from patient records. Integrated LLM-based Q&A for clinical decision support (in progress).

7. Breast Cancer Multi-Output Classification
Built a multi output classification model for breast cancer using iTRAQ proteome profiles of TCGA cancer samples and METABRIC mRNA levels

8. Protein Classification (Deep Learning)
Used protein sequence embeddings (ProtBERT) for functional prediction.

9. TOX24 - Drug Toxicity Prediction (GATs + QSAR)
Applied Graph Attention Networks on molecular structures.

## Open for collaboration
- Email: [shasankashekharpadhi@gmail.com](mailto:shasankashekharpadhi@gmail.com)  
- LinkedIn: [in/shasankashekharpadhi](https://linkedin.com/in/shasankashekharpadhi)  
- Portfolio: [shasanksp.github.io](https://shasanksp.github.io)
